tests/databases/value/mongo_db/test_value_caching.py

```diff
@@ -0,0 +1,48 @@
+import stk
+import pymongo
+
+
+def test_put_caching():
+    collection = '_test_put_caching'
+    database_name = '_test_put_caching'
+    client = pymongo.MongoClient()
+    client.drop_database(database_name)
+
+    database = stk.ValueMongoDb(
+        mongo_client=client,
+        collection=collection,
+        database=database_name,
+    )
+    molecule = stk.BuildingBlock('CCC')
+    database.put(molecule, 43)
+    database.put(molecule, 43)
+
+    cache_info = database._put.cache_info()
+    assert cache_info.hits == 1
+    assert cache_info.misses == 1
+
+    database.put(molecule, 40)
+    cache_info = database._put.cache_info()
+    assert cache_info.hits == 1
+    assert cache_info.misses == 2
+
+
+def test_get_caching():
+    collection = '_test_get_caching'
+    database_name = '_test_get_caching'
+    client = pymongo.MongoClient()
+    client.drop_database(database_name)
+
+    database = stk.ValueMongoDb(
+        mongo_client=client,
+        collection=collection,
+        database=database_name,
+    )
+    molecule = stk.BuildingBlock('CCC')
+    database.put(molecule, 43)
+    database.get(molecule)
+    database.get(molecule)
+
+    cache_info = database._get.cache_info()
+    assert cache_info.hits == 1
+    assert cache_info.misses == 1
```

Comment:
Maybe it's also good to test the cache size in both `put` and `get`? Asserting maxsize value is what we think it is could be useful to check all class arguments are working.

Reply:
The cache size is tested for both `put` and `get` but across different tests. I think that makes more sense as they are different methods and should therefore be tested separately IMO.

I think that somewhat conflates the purpose of the test. The test is meant to purely to test whether caching is taking place. To test if arguments are working should be done in a separate test IMO.